src/components/List/ListItem/index.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { FormEvent, HTMLAttributes, useCallback } from 'react';
+import classNames from 'classnames';
+import { Checkbox } from '../../Checkbox';
+import Icon, { tColors } from '../../Icons';
+import { IControl, IVariant } from '..';
+import Icons from '../../Icons';
+
+type TIcon = {
+  name: string;
+  size?: number;
+  color?: tColors;
+  fill?: boolean;
+} | null;
+
+export interface IListItem extends Omit<HTMLAttributes<HTMLLIElement>, 'onSelect'> {
+  id: string;
+  content: React.ReactNode;
+  name?: string;
+  value?: string;
+  isChecked?: boolean;
+  iconLeft?: TIcon;
+  label?: string;
+  subtext?: string;
+  iconRight?: TIcon;
+  dividers?: 0 | 1 | 2;
+  disabled?: boolean;
+  className?: string;
+}
+
+export interface IListItemExtra extends IListItem {
+  onSelect?: (item: string, evt: React.BaseSyntheticEvent) => void;
+}
+
+export const ListItem: React.FC<IListItemExtra & IVariant & IControl> = (props) => {
+  const {
+    id,
+    name,
+    variant,
+    content,
+    control = 'text',
+    iconLeft = null,
+    subtext = '',
+    label = '',
+    iconRight = null,
+    dividers = 0,
+    disabled = false,
+    isChecked,
+    onSelect,
+    className,
+    ...rest
+  } = props;
+  const handleSelect = useCallback(
+    (evt: React.MouseEvent | FormEvent) => {
+      if (disabled) {
+        return null;
+      }
+
+      onSelect?.(id, evt);
+    },
+    [id, onSelect]
+  );
+
+  const hasIconLeft = !!iconLeft?.name;
+  const hasIconRight = !!((control === 'checkmark' && isChecked) || iconRight?.name);
+
+  return (
+    <li
+      {...rest}
+      className={classNames(className, 'relative items-center bg-background transition-all ease-out duration-300', {
+        ['text-text-disabled']: disabled,
+        ['hover:bg-default-extra-light active:bg-default-light']: !disabled,
+      })}
+    >
+      <button onClick={control !== 'checkbox' ? handleSelect : () => {}} className='flex flex-nowrap w-full text-left'>
```

Comment:
Why do we need the html button component here? By default it has css styles, which we probably have to reset. But I don't really see the point to have button, do we really need it?

Reply:
We have onClick attribute and usually lint rules will scream about it. But I can also change into a span or smth

Reply:
Since there are no default button styles issue it's fine :)